test/extensions/filters/network/ext_proc/ext_proc_integration_test.cc

```diff
@@ -920,6 +920,67 @@ TEST_P(NetworkExtProcFilterIntegrationTest, BothTypedAndUntypedMetadataForwardin
   tcp_client->close();
 }
 
+// Test connection status CLOSE handling in responses
+TEST_P(NetworkExtProcFilterIntegrationTest, ConnectionStatusCloseHandling) {
+  initialize();
+  IntegrationTcpClientPtr tcp_client = makeTcpConnection(lookupPort("network_ext_proc_filter"));
+  ASSERT_TRUE(tcp_client->write("client_data", false));
+
+  FakeRawConnectionPtr fake_upstream_connection;
+  ASSERT_TRUE(fake_upstreams_[0]->waitForRawConnection(fake_upstream_connection));
+
+  ProcessingRequest request;
+  waitForFirstGrpcMessage(request);
+
+  // Create a response with CLOSE status
+  ProcessingResponse response;
+  response.set_connection_status(ProcessingResponse::CLOSE);
+  auto* read_data = response.mutable_read_data();
+  read_data->set_data("modified_data");
+  read_data->set_end_of_stream(false);
+
+  processor_stream_->startGrpcStream();
+  processor_stream_->sendGrpcMessage(response);
+
+  // Verify counters
+  verifyCounters({{"connections_closed", 1}});
+
+  // Connection should be closed
+  ASSERT_FALSE(tcp_client->write("", true));
+  tcp_client->waitForDisconnect();
+}
+
+// Test connection status CLOSE_RST handling in responses
+TEST_P(NetworkExtProcFilterIntegrationTest, ConnectionStatusRSTHandling) {
+  // Create a new HTTP connection since the previous one was closed
+  IntegrationTcpClientPtr tcp_client = makeTcpConnection(lookupPort("network_ext_proc_filter"));
+  ASSERT_TRUE(tcp_client->write("client_data", false));
+
+  FakeRawConnectionPtr fake_upstream_connection;
+  ASSERT_TRUE(fake_upstreams_[0]->waitForRawConnection(fake_upstream_connection));
+
+  ProcessingRequest request;
+  ASSERT_TRUE(grpc_upstream_->waitForHttpConnection(*dispatcher_, processor_connection_));
+  ASSERT_TRUE(processor_connection_->waitForNewStream(*dispatcher_, processor_stream_));
+  ASSERT_TRUE(processor_stream_->waitForGrpcMessage(*dispatcher_, request));
+
+  // Create a response with CLOSE_RST status
+  ProcessingResponse response;
+  response.set_connection_status(ProcessingResponse::CLOSE_RST);
+  auto* read_data = response.mutable_read_data();
+  read_data->set_data("modified_data");
+  read_data->set_end_of_stream(false);
+
+  processor_stream_->startGrpcStream();
+  processor_stream_->sendGrpcMessage(response);
+
+  // Connection should be reset
+  tcp_client->waitForDisconnect();
+
+  // Verify counters - should now have 1 close and 1 reset
```

Comment:
Is the one close from the previous test? If yes then could you add that in the comments?

Reply:
I let the close represent all active close now, and rst close is a specific one, updated the comment.